addons/web/static/tests/fields/basic_fields_tests.js

```diff
@@ -6209,6 +6209,33 @@ QUnit.module('basic_fields', {
         form.destroy();
     });
 
+    QUnit.test('state_selection widget with show_label option', async function (assert) {
+        assert.expect(5);
+
+        const list = await createView({
+            View: ListView,
+            model: 'partner',
+            data: this.data,
+            arch: '<tree>' +
+                    '<field name="foo"/>' +
+                    '<field name="selection" widget="state_selection" options="{\'show_label\': True}"/>' +
+                  '</tree>',
+        });
+
+        assert.containsN(list, '.o_state_selection_cell .o_selection > a span.o_status', 5,
+            "should have five status selection widgets");
+        assert.containsN(list, '.o_state_selection_cell .o_selection > span.align-middle', 5,
+            "should have five label on election widgets");
+        assert.containsOnce(list, '.o_state_selection_cell .o_selection > a span.o_status.o_status_green',
+            "should have one green status");
+        assert.containsOnce(list, '.o_state_selection_cell .o_selection > span.align-middle:contains(Done)',
+            "should have one Done status label");
+        assert.containsN(list, '.o_state_selection_cell .o_selection > span.align-middle:contains(Normal)', 3,
+            "should have one Normal status label");
```

Comment:
Shouldn't we rather only test that the label is displayed when the option is set to True and that the label is not displayed when the option is not used or when it is set to False? By doing so the test is more "unitary".

Reply:
@IT-Ideas , Done all the changes. Thanks